app/com/gu/floodgate/DynamoDBTable.scala

```diff
@@ -111,15 +113,31 @@ trait DynamoDBTable[T] extends StrictLogging {
 
   def updateItem(id: String, t: T): Unit = {
     val request = new UpdateItemRequest().withTableName(tableName)
-      .withKey(Map("id" -> new AttributeValue(id)))
+      .withKey(Map(keyName -> new AttributeValue(id)))
       .withAttributeUpdates(toItemUpdate(t).mapValues(_.withAction("PUT")))
 
     dynamoDB.updateItemAsync(request)
   }
 
-  def deleteItem(id: String) = {
+  /* TODO - implement correctly with return type. */
```

Comment:
It's already returning a `Future[DeleteItemResult]` so you could just change the return type to that? Seems like a reasonable thing to return.